File: src/sub.cpp
#include <iostream>
#include <ros/ros.h>
#include <std_msgs/String.h>

void callback (const std_msgs::String::ConstPtr &msg)
{
	ROS_INFO("subscribe: %s\n", msg->data.c_str());
}


int main (int argc, char **argv)
{
	ros::init(argc, argv, "sub");
	ros::NodeHandle nh;
	ros::Subscriber sub = nh.subscribe("/chatter", 10, callback);
	ros::spin();
	return 0;
}

File: scripts/sub.py
#! /usr/bin/env python

import rospy
from std_msgs.msg import String

def callback (msg):
	rospy.loginfo("subscribe: {0}".format(msg.data))

def main ():
	rospy.init_node("sub", anonymous = True)
	rospy.Subscriber("/chatter", String, callback)
	rospy.spin()

if __name__ == "__main__":
	main()

